### PR TITLE
Added navigation bar

### PR DESCRIPTION
**Pull request by nigow**
### Ticket
https://trello.com/c/aekXS1Vy/5-nav-bar

### What is changed / added
- short description of the PR

Added a navbar along with the MCSS logo

### Screenshots
- if applicable

Screen Shot 2021-08-28 at 1 58 03

### CODE REVIEW

**Comment by TheFatPanda97** [comment]
> @TheFatPanda97 looks like i deleted a bunch of lines from `package-lock.json`. should i revert it?

@nigow could you do a quick merge of master? Package-lock was recently removed so this conflict shouldn't exist

**Comment by TheFatPanda97** [comment]
> @TheFatPanda97 looks like i deleted a bunch of lines from `package-lock.json`. should i revert it?

@nigow yep, let's discard the changes to package-lock.json

**Comment by AaronCGoidel** [comment]
> > @TheFatPanda97 looks like i deleted a bunch of lines from `package-lock.json`. should i revert it?
> 
> @nigow yep, let's discard the changes to package-lock.json

Since we're using `npm` and not `yarn` I think we could do with just `rm`ing the entire lock file.

**Review by AaronCGoidel — COMMENTED** [review]
I think I'm good with all this I'd just like to see screenshots of the mobile view before I pull this in

**Comment by TheFatPanda97** [comment]
> > > @TheFatPanda97 looks like i deleted a bunch of lines from `package-lock.json`. should i revert it?
> > 
> > 
> > @nigow yep, let's discard the changes to package-lock.json
> 
> Since we're using `npm` and not `yarn` I think we could do with just `rm`ing the entire lock file.

I think it will be worth it to keep the lock file. The main benefit of it is so that everyone can have the exact same dependency tree. In the future, if anyone is installing the packages, they would be doing `npm ci` rather than `npm install`.

**Comment by AaronCGoidel** [comment]
> > > > @TheFatPanda97 looks like i deleted a bunch of lines from `package-lock.json`. should i revert it?
> > > 
> > > 
> > > @nigow yep, let's discard the changes to package-lock.json
> > 
> > 
> > Since we're using `npm` and not `yarn` I think we could do with just `rm`ing the entire lock file.
> 
> I think it will be worth it to keep the lock file. The main benefit of it is so that everyone can have the exact same dependency tree. In the future, if anyone is installing the packages, they would be doing `npm ci` rather than `npm install`.

I agree; this is the more professional approach.

I thought that keeping lock files in sync (as we saw in the pr) would be too much to justify the means of staying exactly up to date.

We are meant to have a more professional dev environment in the MCSS.

**Comment by TheFatPanda97** [comment]
@nigow please also fill out the PR descriptions and Trello ticket. This helps the reviewer while reviewing your PR.

**Comment by nigow** [comment]
Sorry, I'm slightly confused with the conversation above. So what is the verdict here?
I assume it is a go with not removing `package-lock.json` from this change https://github.com/utmmcss/mcss-website-frontend/commit/e4f5760bc388ac3452e1d4e2b92f8a274f1e521a but I just want to confirm first.

**Comment by nigow** [comment]
@AaronCGoidel regarding mobile view, I didn't really think about it. here's the mobile view:
Screen Shot 2021-08-29 at 11 44 43
Would you mind merging this first and I can work on mobile view on a separate ticket? The reason is that I want to see the mobile design first. A "collapse" navbar like below would be nice, but we should agree on a design first.
https://stackoverflow.com/questions/48005033/bulma-mobile-layout

**Comment by TheFatPanda97** [comment]
> Sorry, I'm slightly confused with the conversation above. So what is the verdict here?
> I assume it is a go with not removing `package-lock.json` from this change [e4f5760](https://github.com/utmmcss/mcss-website-frontend/commit/e4f5760bc388ac3452e1d4e2b92f8a274f1e521a) (hence my change is OK) but I just want to confirm first.

yep you are exactly right, we won't be removing `package-lock.json`

**Comment by TheFatPanda97** [comment]
@nigow also would you mind merging master? I think there are some conflicts.

**Comment by nigow** [comment]
@TheFatPanda97 rebased

**Comment by AaronCGoidel** [comment]
> view:
> 

Alright, I'm happy saying mobile will be another ticket. This is in spec. You have my go-ahead

File: components/navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import Logo from './../public/mcssLogo.svg'
+import Image from 'next/image'
+
+export const NavBar = (): React.ReactElement => {
```
Comment (TheFatPanda97) [inline components/navbar.tsx:5]: Hmm, the type annotations look a bit odd to me. You probably want to do something like this: 
```
import { FC } from "react";

const NavBar: FC = ({
  title,
  content,
}) => {
  body ...
}

export default NavBar;
```

Sidenote, the convention for our project will be to create the const component first then export default at the bottom.

File: components/navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    return (
+        <>
+            <nav className={"flex items-center justify-between flex-wrap bg-teal-500 mt-14 ml-24 mr-24"}>
+                <div className={"flex item-center flex-shrink-0 mr-6"}>
+                    <Image src={Logo} alt={"MCSS logo"}></Image>
+                </div>
+                <div className={"w-full block flex-grow lg:flex lg:items-center lg:w-auto"}>
```
Comment (TheFatPanda97) [inline components/navbar.tsx:12]:
```suggestion
                <div className="w-full block flex-grow lg:flex lg:items-center lg:w-auto">
```
This also appears in a few other places

**Review by TheFatPanda97 — CHANGES_REQUESTED** [review]
Looks pretty solid 👍  Just a few small issue with types and styling. Also, definitely merge master since there has been a lot of linting and type checks added since the creation of this branch.

File: components/navbar.tsx (hunk 2 above)
Comment (nigow) [reply components/navbar.tsx]: Fixed, thanks

File: components/navbar.tsx (hunk 1 above)
Comment (nigow) [reply components/navbar.tsx]: I'm going to follow the type annotation here, but not really sure if that even matters in small components like this which requires no props.

Noted on `export default` convention.

**Review by TheFatPanda97 — APPROVED** [review]
LGTM!

**Comment by TheFatPanda97** [comment]
[Nav bar](https://trello.com/c/aekXS1Vy/5-nav-bar)

**Pull request merged**